<launch>
	<remap from="/output/sbus/cmd_vel" to="/cmd_vel" />

	<node pkg="sbus_serial" type="sbus_serial_node" name="sbus_node">
		<param name="port" value="/dev/ttyTHS2" />
		<param name="refresh_rate_hz" value="10" />
		<param name="rxMinValue" value="172" />
		<param name="rxMaxValue" value="1811" />
		<param name="outMinValue" value="-100" />
		<param name="outMaxValue" value="100" />
		<param name="silentOnFailsafe" value="true" />

		<!-- To only publish messages when a specific channel is within a certain range, use the following three parameters -->
		<!--
		The enableChannelNum is the channel number to check. Note, this is the channel number, not the array index. So channel 1 is "1" and not "0".
		If enableChannelNum is set to -1 or 0 or higher than 16, no check is performed.
		Set enableChannelProportionalMin and enableChannelProportionalMax to the "proportional values" within which publishing is enabled.
		Example: to publish only if channel 5 is above 50%:
			<param name="enableChannelNum" value="5" />
			<param name="enableChannelProportionalMin" type="double" value="0.5" />
			<param name="enableChannelProportionalMax" type="double" value="1.0" />
		-->
		<param name="enableChannelNum" value="6" />
		<param name="enableChannelProportionalMin" type="double" value="0.5" />
		<param name="enableChannelProportionalMax" type="double" value="1.0" />

	</node>
	<node pkg="sbus_serial" type="sbus_cmd_vel_node" name="sbus_cmd_vel">
		<param name="sbusMinValue" value="-100" />
		<param name="sbusMaxValue" value="100" />
		<param name="maxSpeed" value="1.8" />
		<param name="maxTurn" value="18" />
	</node>
</launch>
